### PR TITLE
Add readDefaultConfig()

### PR DESCRIPTION
**Pull request by mikapfl**
Hi,

This adds the `readDefaultConfig()` function to consistently read a model's default config from the right place. It was surprisingly hard to get right.

Note that the function does *not* preserve the order of settings. Is that a problem?

Cheers

Mika

### CODE REVIEW

File: R/readDefaultConfig.R [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#' @return A vector of parameter values and their names.
+#' @author Mika Pflüger
+#' @importFrom withr with_environment
+#' @importFrom utils modifyList
+#' @export
+
+readDefaultConfig <- function(path) {
+  # read in settings from main.gms first
+  cfg <- list()
+  cfg$gms <- as.list(readSettings(paste0(path, "/", "main.gms")))
```
Comment (LaviniaBaumstark) [inline R/readDefaultConfig.R:16]: this should be the default but I guess, we need more flexibility for stand alone models

File: R/readDefaultConfig.R [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  # overwrite with settings from default.cfg
+  env <- new.env()
+  source(paste0(path, "/", "config/default.cfg"), local = env)  # nolint: undesirable_function_linter
```
Comment (LaviniaBaumstark) [inline R/readDefaultConfig.R:19]: see comment regarding main.gms

File: R/readDefaultConfig.R (hunk 1 above)
Comment (mikapfl) [reply R/readDefaultConfig.R:16]: Hm, what is the use case you are missing? When people have a model which does not have a `main.gms`?

Comment (LaviniaBaumstark) [reply R/readDefaultConfig.R:16]: we already have a mofex.gms and trade.gms for those stand-alone modles

Comment (mikapfl) [reply R/readDefaultConfig.R:18]: I think we can add functionality for stand-alone models when we need it.

**Review by LaviniaBaumstark — APPROVED** [review]

**Pull request merged**